packages/core/src/messagingProtocols/mls/E2EIdentityService/Steps/Authorization.ts

```diff
@@ -20,46 +20,55 @@
 import {AcmeService} from '../Connection';
 import {E2eiEnrollment, NewAcmeAuthz, Nonce} from '../E2EIService.types';
 import {jsonToByteArray} from '../Helper';
+import {AuthData} from '../Storage/E2EIStorage.schema';
 
 interface GetAuthorizationParams {
   nonce: Nonce;
-  authzUrl: string;
+  authzUrls: string[];
   identity: E2eiEnrollment;
   connection: AcmeService;
 }
-export type AuthorizationChallenge = {authorization: NewAcmeAuthz; nonce: Nonce};
 
 export const getAuthorizationChallenges = async ({
-  authzUrl,
+  authzUrls,
   nonce,
   identity,
   connection,
-}: GetAuthorizationParams): Promise<AuthorizationChallenge> => {
-  const reqBody = await identity.newAuthzRequest(authzUrl, nonce);
-  const response = await connection.getAuthorization(authzUrl, reqBody);
+}: GetAuthorizationParams): Promise<AuthData> => {
+  const challenges: {type: string; challenge: NewAcmeAuthz}[] = [];
 
-  if (response?.data && !!response.data.status.length && !!response.nonce.length) {
-    const wasmData = await identity.newAuthzResponse(jsonToByteArray(response.data));
-    // manual copy of the wasm data because of a problem while cloning it
-    const authorization: NewAcmeAuthz = {
-      identifier: wasmData.identifier,
-      keyauth: wasmData.keyauth,
-      wireDpopChallenge: {
-        delegate: wasmData.wireDpopChallenge!.delegate,
-        target: wasmData.wireDpopChallenge!.target,
-        url: wasmData.wireDpopChallenge!.url,
-      },
-      wireOidcChallenge: {
-        delegate: wasmData.wireOidcChallenge!.delegate,
-        target: wasmData.wireOidcChallenge!.target,
-        url: wasmData.wireOidcChallenge!.url,
-      },
-    };
-    return {
-      authorization,
-      nonce: response.nonce,
-    };
+  for (const authzUrl of authzUrls) {
+    const reqBody = await identity.newAuthzRequest(authzUrl, nonce);
+    const response = await connection.getAuthorization(authzUrl, reqBody);
+    const backendChallenge = response.data.challenges[0];
+    const challenge = await identity.newAuthzResponse(jsonToByteArray(response.data));
+    challenges.push({type: backendChallenge.type, challenge});
```

Comment:
This is the main change. Now the `order` gives us back 2 urls, and we need to create 2 challenges from CoreCrypto with those 2 urls